packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding_phone_confirmation/blocs/onboarding_phone_confirm_bloc.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:rx_bloc/rx_bloc.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../base/common_services/onboarding_service.dart';
+import '../../base/models/user_model.dart';
+import '../../lib_permissions/services/permissions_service.dart';
+import '../../lib_router/blocs/router_bloc.dart';
+import '../../lib_router/router.dart';
+
+part 'onboarding_phone_confirm_bloc.rxb.g.dart';
+
+abstract class OnboardingPhoneConfirmBlocEvents {
+  /// Sets the confirmation result
+  void setConfirmationResult(dynamic result);
+}
+
+abstract class OnboardingPhoneConfirmBlocStates {
+  /// The routing state for navigating to next page
+  ConnectableStream<void> get onRouting;
+}
+
+@RxBloc()
+class OnboardingPhoneConfirmBloc extends $OnboardingPhoneConfirmBloc {
+  OnboardingPhoneConfirmBloc(
+    this._routerBloc,
+    this._onboardingService,
+    this._permissionsService,
+  ) {
+    onRouting.connect().addTo(_compositeSubscription);
+  }
+
+  final OnboardingService _onboardingService;
+  final RouterBlocType _routerBloc;
+  final PermissionsService _permissionsService;
+
+  @override
+  ConnectableStream<void> _mapToOnRoutingState() =>
+      _$setConfirmationResultEvent.map(_onResultChanged).publish();
```

Comment:
Please use the async operators such as asyncMap, switchMap, flatMap etc for async functions. Also do not forget to handle potential errors